crates/subspace-farmer/src/commitments/commitment_databases.rs

```diff
@@ -95,62 +103,125 @@ impl CommitmentDatabases {
         }
 
         // Open databases that were fully created during previous run
-        for salt in commitment_databases.metadata_cache.keys() {
-            let db = DB::open(&Options::default(), base_directory.join(hex::encode(salt)))
-                .map_err(CommitmentError::CommitmentDb)?;
-            commitment_databases.databases.put(
-                *salt,
-                Arc::new(DbEntry {
-                    salt: *salt,
+        let read_lock = commitment_databases.metadata.read();
+        let first_salt = read_lock.metadata_cache.keys().next();
+        let second_salt = read_lock.metadata_cache.keys().next();
+
+        // Open the first one
+        if let Some(first_salt) = first_salt {
+            let db = DB::open(
+                &Options::default(),
+                base_directory.join(hex::encode(first_salt)),
+            )
+            .map_err(CommitmentError::CommitmentDb)?;
+            commitment_databases
+                .first_db
+                .write()
+                .replace(Arc::new(DbEntry {
+                    salt: *first_salt,
                     db: Mutex::new(Some(Arc::new(db))),
-                }),
-            );
+                }));
         }
+        // Open the second one
+        if let Some(second_salt) = second_salt {
+            let db = DB::open(
+                &Options::default(),
+                base_directory.join(hex::encode(second_salt)),
+            )
+            .map_err(CommitmentError::CommitmentDb)?;
+            commitment_databases
+                .second_db
+                .write()
+                .replace(Arc::new(DbEntry {
+                    salt: *second_salt,
+                    db: Mutex::new(Some(Arc::new(db))),
+                }));
+        }
+        drop(read_lock);
 
         Ok::<_, CommitmentError>(commitment_databases)
     }
 
     /// Get salts for all current database entries
     pub(super) fn get_salts(&self) -> Vec<Salt> {
-        self.databases
+        self.metadata
+            .read()
+            .metadata_cache
             .iter()
-            .map(|(salt, _db_entry)| *salt)
+            .map(|(salt, _commitment_status)| *salt)
             .collect()
     }
 
-    pub(super) fn get_db_entry(&self, salt: &Salt) -> Option<&Arc<DbEntry>> {
-        self.databases.peek(salt)
+    pub(super) fn get_db_entry(&self, salt: &Salt) -> Option<Arc<DbEntry>> {
+        return match salt {
+            _ if &self.first_db.read().as_ref().unwrap().salt == salt => {
```

Comment:
this looks ugly in my opinion. But I don't have if there is a more elegant way...